src/foam/nanos/crunch/Renewable.js

```diff
@@ -7,123 +7,333 @@
 
   javaImports: [
+    'foam.core.ContextAware',
+    'foam.core.PropertyInfo',
+    'foam.core.X',
+    'foam.core.XLocator',
+    'foam.dao.DAO',
+    'static foam.mlang.MLang.EQ',
+    'static foam.mlang.MLang.OR',
+    'foam.nanos.logger.Logger',
+    'foam.nanos.logger.Loggers',
+    'foam.time.TimeZone',
+    'foam.util.SafetyUtil',
+    'java.time.*',
+    'java.time.temporal.*',
     'java.util.Date'
   ],
 
+  sections: [
+    { name: 'renewableSection' }
+  ],
+
   properties: [
     {
-      name: 'isExpired',
-      class: 'Boolean',
-      storageTransient: true,
-      javaGetter: `
-      if ( getExpiry() == null ) return false;
-
-      Date today = new Date();
-      Date capabilityExpiry = getExpiry();
-
-      return today.after(capabilityExpiry);
-      `
+      class: 'Reference',
+      of: 'foam.time.TimeZone',
+      name: 'timeZone',
+      order: 0,
+      value: 'Africa/Abidjan', // UTC/GMT
+      section: 'renewableSection'
     },
     {
-      name: 'isRenewable',
-      class: 'Boolean',
-      getter: function() { return this.isExpired || this.isInRenewablePeriod || this.isInGracePeriod; },
-      javaGetter: `
-        return getIsExpired() || getIsInRenewablePeriod() || getIsInGracePeriod();
-      `
-    },
-    {
-      name: 'isInRenewablePeriod',
-      class: 'Boolean',
+      name: 'expiry',
+      class: 'DateTime',
+      documentation: `Datetime of when capability is no longer valid. Explicitly set, or calculated from period and periodTimeUnit`,
+      includeInDigest: true,
+      section: 'renewableSection',
       javaSetter: `
-        isInRenewablePeriod_ = val;
-        isInRenewablePeriodIsSet_ = true;
-        if ( isInRenewablePeriod_ ) {
-          isExpired_ = false;
-          isInGracePeriod_ = false;
-        }
+      expiry_ = val;
+      expiryIsSet_ = true;
+      reset();
       `
     },
     {
-      name: 'isInGracePeriod',
-      class: 'Boolean',
+      name: 'expiryPeriod',
+      class: 'Int',
+      documentation: `The number of TimeUnits to calculate expiry date.
+      NOTE: existing logic - The UserCapabilityJunction object will have its expiry configured to a DateTime based on the lower value of the two, expiry and duration`,
+      aliases: ['duration'],
+      includeInDigest: true,
+      section: 'renewableSection',
       javaSetter: `
-        isInGracePeriod_ = val;
-        isInGracePeriodIsSet_ = true;
-        if ( isInGracePeriod_ ) {
-          isExpired_ = false;
-          isInRenewablePeriod_ = false;
-        }
+      expiryPeriod_ = val;
+      expiryPeriodIsSet_ = true;
+      reset();
       `
     },
     {
-      name: 'expiry',
-      class: 'DateTime',
-      documentation: `Datetime of when capability is no longer valid. Explicitly set, or calculated from duration and durationTimeUnit`,
+      documentation: 'Unit of Duration',
+      name: 'expiryPeriodTimeUnit',
+      class: 'Enum',
+      of: 'foam.time.TimeUnit',
+      value: 'DAY',
```

Comment:
should this be year?

Reply:
No, it's just [this that defaults to YEAR](https://github.com/kgrgreer/foam3/pull/2914/files#diff-8a96cd310c262b31d2642edcdf6f6e32f67181bf0e9dbf6ebd4908409974dac4R31)